### ISSUE TITLE
eval_loss is always nan during lora fine tuning qwen2.5-vl-3b-instruct

### ISSUE THREAD

**Issue by doggnol** [modelscope/ms-swift#5580]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)

I was lora finetuning qwen-2.5-vl-3b-instruct on v100, my training_args are as follows:

```python
# lora
lora_rank = 8
lora_alpha = 32
lora_dropout = 0.05
freeze_llm = False
freeze_vit = True
freeze_aligner = True

# training_args
training_args = Seq2SeqTrainingArguments(
    output_dir=output_dir,
    learning_rate=5e-5,
    per_device_train_batch_size=1,
    per_device_eval_batch_size=1,
    gradient_checkpointing=True,
    fp16=True,
    weight_decay=0.01,
    lr_scheduler_type='cosine',
    warmup_ratio=0.03,
    report_to=['tensorboard'],
    logging_first_step=True,
    save_strategy='steps',
    eval_strategy='steps',
    eval_steps=20,
    save_steps=20,
    logging_steps=5,
    load_best_model_at_end=True,
    metric_for_best_model="eval_loss",
    greater_is_better=False,
    gradient_accumulation_steps=8,
    # To observe the training results more quickly, this is set to 1 here. 
    # Under normal circumstances, a larger number should be used.
    num_train_epochs=10,
    save_total_limit=5,
    dataloader_num_workers=0,
    data_seed=data_seed,
    remove_unused_columns=False,
    dataloader_pin_memory=True,
    label_names=["labels"],
    deepspeed="/home/vlm/notebook/ds_config.json"
)
```

the training loss is normal, but eval_loss is always nan, which makes earlystopping not work. 

logs are as follows:

```
{'loss': 0.59159303, 'token_acc': 0.85819352, 'grad_norm': 0.54187453, 'learning_rate': 2.27e-06, 'memory(GiB)': 27.79, 'train_speed(iter/s)': 0.018369, 'epoch': 0.28, 'global_step/max_steps': '1/720', 'percentage': '0.14%', 'elapsed_time': '50s', 'remaining_time': '10h 9m 47s'}
{'eval_loss': nan, 'eval_runtime': 175.8374, 'eval_samples_per_second': 1.16, 'eval_steps_per_second': 0.58, 'epoch': 0.28, 'global_step/max_steps': '1/720', 'percentage': '2.78%', 'elapsed_time': '17m 16s', 'remaining_time': '10h 4m 43s'} 
```

I also tried qwen2-vl-2b-instruct, the eval_loss is normal.

logs are here:

```
{'loss': 0.4522028, 'token_acc': 0.88169343, 'grad_norm': 0.40547693, 'learning_rate': 4.545e-05, 'memory(GiB)': 27.82, 'train_speed(iter/s)': 0.025985, 'epoch': 0.28, 'global_step/max_steps': '20/720', 'percentage': '2.78%', 'elapsed_time': '12m 46s', 'remaining_time': '7h 26m 54s'}
Train:   3%|█████▋                                                                                                                                                                                                      | 20/720 [12:46<6:04:53, 31.28s/it]
{'eval_loss': 0.41330269, 'eval_runtime': 117.0905, 'eval_samples_per_second': 1.742, 'eval_steps_per_second': 0.871, 'epoch': 0.28, 'global_step/max_steps': '20/720', 'percentage': '2.78%', 'elapsed_time': '14m 43s', 'remaining_time': '8h 35m 12s'}
```

Does any one has the same problem？

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)

CUDA version: 12.2
system: ubuntu 22.04 x86_64
gpu: v100-32g-pcie * 2
torch version: 2.6.0

**Additional context**
Add any other context about the problem here(在这里补充其他信息)

**Comment by Jintao-Huang** [comment]
使用命令行训练试试

**Comment by doggnol** [comment]
我尝试了命令行，并且因此改动了一些deepspeed的参数，eval_loss还是nan

这是我使用的命令行：

```shell
CUDA_VISIBLE_DEVICES=0,1 \
ACCELERATE_USE_FSDP=true \
MAX_PIXELS=1003520 \
# swift sft \
torchrun \
    --nproc_per_node=2 \
    -m swift.cli.sft \
    --model /home/vlm/model/qwen2.5-vl-3b-instruct \
    --dataset '/home/vlm/datasets/test.jsonl' \
    --output_dir output \
    --learning_rate 5e-5 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 1 \
    --gradient_checkpointing true \
    --fp16 true \
    --weight_decay 0.01 \
    --lr_scheduler_type cosine \
    --warmup_ratio 0.03 \
    --logging_first_step true \
    --save_strategy steps \
    --save_steps 20 \
    --save_total_limit 3 \
    --eval_strategy steps \
    --eval_steps 1 \
    --logging_steps 5 \
    --load_best_model_at_end true \
    --metric_for_best_model eval_loss \
    --greater_is_better false \
    --gradient_accumulation_steps 8 \
    --num_train_epochs 10 \
    --dataloader_num_workers 4 \
    --split_dataset_ratio 0.15 \
    --data_seed 42 \
    --dataloader_pin_memory true \
    --label_names labels \
    --train_type lora \
    --torch_dtype float16 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --lora_dropout 0.05 \
    --freeze_vit true \
    --freeze_aligner true \
    --freeze_llm false \
    --max_length 8192 \
    --deepspeed /home/vlm/notebook/ds_config.json
```

/home/vlm/notebook/ds_config.json

```json
{
    "train_micro_batch_size_per_gpu": 1,
    "gradient_accumulation_steps": 8,
    "optimizer": {
        "type": "AdamW",
        "params": {
            "lr": 5e-5,
            "weight_decay": 0.01,
            "betas": [0.9, 0.95]
        }
    },
    "zero_optimization": {
        "stage": 2,
        "allgather_partitions": true,
        "allgather_bucket_size": 5e8,
        "overlap_comm": true,
        "reduce_scatter": true,
        "reduce_bucket_size": 5e8,
        "contiguous_gradients": true
    },
    "fp16": {
        "enabled": true
    },
    "bf16": {
        "enabled": false
    },
    "gradient_clipping": 1.0,
    "flops_profiler": {
        "enabled": false
    }
}
```

训练日志如下：

```
{'loss': 0.4544507, 'token_acc': 0.88763657, 'grad_norm': 0.3002567, 'learning_rate': 2.08e-06, 'memory(GiB)': 30.88, 'train_speed(iter/s)': 0.009925, 'epoch': 0.01, 'global_step/max_steps': '1/800', 'percentage': '0.12%', 'elapsed_time': '1m 1s', 'remaining_time': '13h 45m 19s'}
Train:   0%|▎                                                                                                                                                                                                           | 1/800 [01:01<13:45:16, 61.97s/it]
{'eval_loss': nan, 'eval_token_acc': 0.86867758, 'eval_runtime': 54.8588, 'eval_samples_per_second': 1.24, 'eval_steps_per_second': 0.62, 'epoch': 0.01, 'global_step/max_steps': '1/800', 'percentage': '0.12%', 'elapsed_time': '1m 56s', 'remaining_time': '1d 1h 55m 57s'}
Val: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 34/34 [00:52<00:00,  1.54s/it]
```

**Comment by doggnol** [comment]
更新，我似乎找到了问题，将split_dataset_ratio设置为0.01后，验证集数量来到7，此时eval_loss不再是nan了。

日志如下：

```
{'loss': 0.47545946, 'token_acc': 0.87098592, 'grad_norm': 0.33568189, 'learning_rate': 1.92e-06, 'memory(GiB)': 29.69, 'train_speed(iter/s)': 0.02425, 'epoch': 0.01, 'global_step/max_steps': '1/840', 'percentage': '0.12%', 'elapsed_time': '38s', 'remaining_time': '9h 3m 31s'}
Train:   0%|▏                                                                                                                                                                                                            | 1/840 [00:38<9:03:28, 38.87s/it]
{'eval_loss': 0.44413444, 'eval_token_acc': 0.89439778, 'eval_runtime': 11.6649, 'eval_samples_per_second': 1.114, 'eval_steps_per_second': 0.6, 'epoch': 0.01, 'global_step/max_steps': '1/840', 'percentage': '0.12%', 'elapsed_time': '50s', 'remaining_time': '11h 46m 44s'}
Val: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:10<00:00,  1.43s/it]
```

但是如果我将split_dataset_ratio增到到0.15，现象又会出现。

**Comment by doggnol** [comment]
感谢 @Jintao-Huang

**Comment by doggnol** [comment]
> 更新，我似乎找到了问题，将split_dataset_ratio设置为0.01后，验证集数量来到7，此时eval_loss不再是nan了。
> 
> 日志如下：
> 
> ```
> {'loss': 0.47545946, 'token_acc': 0.87098592, 'grad_norm': 0.33568189, 'learning_rate': 1.92e-06, 'memory(GiB)': 29.69, 'train_speed(iter/s)': 0.02425, 'epoch': 0.01, 'global_step/max_steps': '1/840', 'percentage': '0.12%', 'elapsed_time': '38s', 'remaining_time': '9h 3m 31s'}
> Train:   0%|▏                                                                                                                                                                                                            | 1/840 [00:38<9:03:28, 38.87s/it]
> {'eval_loss': 0.44413444, 'eval_token_acc': 0.89439778, 'eval_runtime': 11.6649, 'eval_samples_per_second': 1.114, 'eval_steps_per_second': 0.6, 'epoch': 0.01, 'global_step/max_steps': '1/840', 'percentage': '0.12%', 'elapsed_time': '50s', 'remaining_time': '11h 46m 44s'}
> Val: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 7/7 [00:10<00:00,  1.43s/it]
> ```
> 
> 但是如果我将split_dataset_ratio增到到0.15，现象又会出现。

这提醒我可能是验证集的数据有问题，导致loss计算出现nan，我通过在基础模型上计算loss的方式找到了有问题的数据，但是暂时还没有看出来数据有啥问题。

我准备先把问题数据删掉，然后跑起来训练试试看。

**Comment by doggnol** [comment]
更新，确定了是数据的问题，把问题数据过滤掉以后，eval_loss可以正常计算了。

话说训练集也有问题数据，但是训练集算loss是没问题的，不知道是不是对异常数据做了特殊处理，而验证集没加。
